File: ringBuffer.js
class Buffer {
  constructor(capacity) {
    this.capacity = capacity,
    this.storage = [],
    this.current = 0   
  }
  append(item) {
    this.storage[this.current++] = item
    if (this.current === this.capacity) {
      this.current = 0
    }
  }
  get(item) {
    return this.storage
  }
}

File: ringBuffer.py
class RingBuffer:

  def __init__(self, capacity):
    self.capacity = capacity
    self.current = 0
    self.storage = []

  def append(self, item):
    self.storage[self.current++] = item

    if self.current = self.capacity:
      self.current = 0

  def get():
    return self.storage